### PR TITLE
Fix Table.Rows Rendering Error for One Child

### PR DESCRIPTION
**Pull request by emilyrohrbough**
### Summary
Updated Rows to handle zero-many children by using React.Children.map.
Addresses #254 

@cerner/terra

### CODE REVIEW

File: packages/terra-table/src/TableRows.jsx [hunk 1]
```diff
@@ -23,7 +23,7 @@ const defaultProps = {
 };
 
 function cloneChildItems(children, onClick, onKeyDown) {
-  return children.map((child) => {
+  return React.Children.map(children, (child) => {
```
Comment (StephenEsser) [inline packages/terra-table/src/TableRows.jsx]: An alternative is to mark children as required and or default children to an empty array
```
const defaultProps = {
  children: [],
  onClick: undefined,
  onKeyDown: undefined,
};
```

**Comment by Matt-Butler** [comment]
@emilyrohrbough If you have time, can you add a test so we can verify this change fixed the issue?

**Comment by emilyrohrbough** [comment]
@Matt-Butler Tests included.

**Comment by Matt-Butler** [comment]
@emilyrohrbough Thank you!

**Pull request merged**